components/blitz/src/loci/ome/io/OmeroReader.java

```diff
@@ -166,25 +177,43 @@ public boolean isThisType(String name, boolean open) {
         final int[] zct = FormatTools.getZCTCoords(this, no);
 
         final byte[] plane;
+        RawPixelsStorePrx store = null;
         try {
+            store = serviceFactory.createRawPixelsStore();
+            store.setPixelsId(pix.getId().getValue(), false);
             plane = store.getPlane(zct[0], zct[1], zct[2]);
         }
         catch (ServerError e) {
             throw new FormatException(e);
+        } finally {
+            // cannot use try-with-resources
+            // RawPixelsStorePrx does not implement autocloseable
+            if (store != null) {
+                try {
+                    store.close();
+                } catch (Exception ex) {
+                    throw new FormatException(ex);
+                }
+            }
         }
 
-        RandomAccessInputStream s = new RandomAccessInputStream(plane);
-        readPlane(s, x, y, w, h, buf);
-        s.close();
+        try (RandomAccessInputStream s = new RandomAccessInputStream(plane)) {
+            readPlane(s, x, y, w, h, buf);
+        }
 
         return buf;
     }
 
     @Override
     public void close(boolean fileOnly) throws IOException {
         super.close(fileOnly);
-        if (!fileOnly && client != null) {
-            client.closeSession();
+        if (!fileOnly) {
```

Comment:
Understood. I thought we established during work on #4817 that now we don't need that?